pkg/nfd-master/nfd-master.go

```diff
@@ -970,18 +985,37 @@ func (m *nfdMaster) updateNodeObject(cli *kubernetes.Clientset, nodeName string,
 			extendedResourceKeys = append(extendedResourceKeys, strings.TrimPrefix(key, nfdv1alpha1.FeatureLabelNs+"/"))
 		}
 		sort.Strings(extendedResourceKeys)
-		annotations[m.instanceAnnotation(nfdv1alpha1.ExtendedResourceAnnotation)] = strings.Join(extendedResourceKeys, ",")
+		nfdAnnotations[m.instanceAnnotation(nfdv1alpha1.ExtendedResourceAnnotation)] = strings.Join(extendedResourceKeys, ",")
+	}
+
+	// Store names of annotations in an annotation
+	annotationKeys := make([]string, 0, len(featureAnnotations))
+	for key := range featureAnnotations {
+		// Drop the ns part for annotations in the default ns
+		annotationKeys = append(annotationKeys, strings.TrimPrefix(key, nfdv1alpha1.FeatureAnnotationNs+"/"))
+	}
+	sort.Strings(annotationKeys)
+	nfdAnnotations[m.instanceAnnotation(nfdv1alpha1.NodeFeatureAnnotation)] = strings.Join(annotationKeys, ",")
+
+	annotations := make(Annotations)
+	for k, v := range nfdAnnotations {
+		annotations[k] = v
+	}
+	for k, v := range featureAnnotations {
+		annotations[k] = v
 	}
 
 	// Create JSON patches for changes in labels and annotations
 	oldLabels := stringToNsNames(node.Annotations[m.instanceAnnotation(nfdv1alpha1.FeatureLabelsAnnotation)], nfdv1alpha1.FeatureLabelNs)
+	oldAnnotations := stringToNsNames(node.Annotations[m.instanceAnnotation(nfdv1alpha1.NodeFeatureAnnotation)], nfdv1alpha1.FeatureAnnotationNs)
 	patches := createPatches(oldLabels, node.Labels, labels, "/metadata/labels")
-	patches = append(patches,
-		createPatches(
-			[]string{nfdv1alpha1.FeatureLabelsAnnotation, nfdv1alpha1.ExtendedResourceAnnotation},
-			node.Annotations,
-			annotations,
-			"/metadata/annotations")...)
+	//patches = append(patches,
+	//	createPatches(
+	//		[]string{nfdv1alpha1.FeatureLabelsAnnotation, nfdv1alpha1.ExtendedResourceAnnotation},
+	//		node.Annotations,
+	//		annotations,
+	//		"/metadata/annotations")...)
+	patches = append(patches, createPatches(oldAnnotations, node.Annotations, annotations, "/metadata/annotations")...)
```

Comment:
Need to append `{nfdv1alpha1.FeatureLabelsAnnotation, nfdv1alpha1.ExtendedResourceAnnotation}` from the deleted code to oldAnnotations here